addons/assignGear/functions/fnc_addSupplyBoxActions.sqf

```diff
@@ -0,0 +1,94 @@
+#include "script_component.hpp"
+/*
+ * Author: Lambda.Tiger
+ * Adds actions to allow a client to mark a supply box using smoke, glow sticks
+ * and or map markers. Smoke and glow sticks are enabled with a marking level
+ * of 1 or greater, and map markers at level 2 or greater.
+ *
+ * Arguments:
+ * 0: Box <OBJECT>
+ * 1: Marking level <SCALAR>
+ *    >=1 allows smoke and chem lights to be added
+ *    >=2 allows for a map marker to be placed/updated
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorObject, 2] call potato_assignGear_fnc_addSupplyBoxActions.sqf
+ *
+ * Public: No
+ */
+#define LARGE_UNIQUE_MARKER_INDEX 934648
+params ["_theBox", "_markingLevel"];
+
+if (_markingLevel < 1 || !hasInterface) exitWith {};
+
+private _action = [
+    "markBoxBase",
+    "Mark Box",
+    "\a3\ui_f\data\igui\cfg\simpletasks\types\move_ca.paa", {
+        private _marker = _target getVariable [QGVAR(boxMarker), ""];
+        if (getMarkerPos _marker isNotEqualTo [0, 0, 0]) then {
+            _marker setMarkerPos (getPosATL _target);
+        };
+    },
+    {true}
+] call ACEFUNC(interact_menu,createAction);
+[_theBox, 0, ["ACE_MainActions"], _action] call ACEFUNC(interact_menu,addActionToObject);
+
+_action = [
+    "markBoxSmoke",
+    "Smoke",
+    "\a3\Modules_F_Curator\Data\portraitSmoke_ca.paa", {
+    private _smoke = createVehicle ["SmokeShellYellow", ASLToAGL getPosASL _target, [], 0, "CAN_COLLIDE"];
+    _smoke attachTo [_target, [0, 0, 0]];
+    _target setVariable [QGVAR(smokeAvailable), false];
+    },
+    {_target getVariable [QGVAR(smokeAvailable), true]}
+] call ACEFUNC(interact_menu,createAction);
+[_theBox, 0, ["ACE_MainActions","markBoxBase"], _action] call ACEFUNC(interact_menu,addActionToObject);
+_action = [
+    "markBoxGlowstick",
+    "Glow Stick",
+    "\a3\Modules_F_Curator\Data\portraitChemlight_ca.paa", {
+    private _smoke = createVehicle ["ACE_G_Chemlight_HiYellow", ASLToAGL getPosASL _target, [], 0, "CAN_COLLIDE"];
+    _smoke attachTo [_target, [0, 0, 0]];
+    _target setVariable [QGVAR(glowStickAvailable), false];
+    },
+    {_target getVariable [QGVAR(glowStickAvailable), true]}
+] call ACEFUNC(interact_menu,createAction);
+[_theBox, 0, ["ACE_MainActions","markBoxBase"], _action] call ACEFUNC(interact_menu,addActionToObject);
+
+if (_markingLevel < 2) exitWith {};
+_action = [
+    "markBoxMapMarker",
+    "Map Marker",
+    "\a3\ui_f\data\igui\cfg\simpletasks\types\rearm_ca.paa", {
+        private _marker = _target getVariable [QGVAR(boxMarker), ""];
+        // when a marker name/values are edit, the marker is re-created so it's possible to lose it (often)
+        if (getMarkerPos _marker isEqualTo [0, 0, 0]) then {
+            _marker = createMarkerLocal [
```

Comment:
Q: this is fine as is, but does @kil0byt3 want local or global marking?

Reply:
it should probably be `createMarker` bc idk how local works with JIP (probably not as expected)

Reply:
local works as in: only visible to the person marking it

Reply:
global it is

Reply:
do you want side specific markers? how will it work with “joint ops” (blufor and indie same team coop). same with tvts

Reply:
I changed how those are ordered, it used to end with a non-local command to finally broadcast it, nice catch.

Reply:
global + side channel ideally, joint ops they'll just have to suck it or re-mark the map, I guess (unless you wanted to make it check if sides were friendly to blufor and mark it in their side channel too...that's a lot more complex).

 feel like this has a higher probability of being used in a TVT than in a blu/indy same team coop, and we should create it for that possibility more than a joint op. but if there's a better way to do it, then go with that

Reply:
You are right, I do default to side there, it could be global either by default OR done by logi once they place it